src/Lean/MetavarContext.lean

```diff
@@ -588,8 +588,15 @@ def instantiateMVars [Monad m] [MonadMCtx m] (e : Expr) : m Expr := do
     return r
 
 def instantiateLCtxMVars [Monad m] [MonadMCtx m] (lctx : LocalContext) : m LocalContext :=
+  let auxDeclToFullName := lctx.auxDeclToFullName
   lctx.foldlM (init := {}) fun lctx ldecl => do
      match ldecl with
+     | .cdecl _ fvarId userName type _ .auxDecl =>
+        let type ← instantiateMVars type
+        let .some fullName := auxDeclToFullName.find? fvarId
+          | panic! s!"Invalid auxiliary declaration found in local context: \
+                      {userName} does not have an associated full name."
+        return lctx.mkAuxDecl fvarId userName type fullName
```

Comment:
Seeing this code I wonder if `Lean.LocalDeclKind.auxDecl` could carry the full name information, instead of having a separate map. But probably better to not obsess much more about this refactoring, there are always many ways to do it. We can reconsider when and if the current approach shows issues.